streams/client/client.go

```diff
@@ -234,64 +263,87 @@ func (c *Client) connect(peer peerCore.AddrInfo) (network.Stream, bool, error) {
 	return strm, false, nil
 }
 
-func (c *Client) sendTo(strm stream, deadline time.Time, cmdName string, body command.Body) response {
+func (c *Client) sendTo(strm stream, deadline time.Time, cmdName string, body command.Body) *Response {
 	cmd := command.New(cmdName, body)
 
 	if err := strm.SetWriteDeadline(min(time.Now().Add(SendTimeout), deadline)); err != nil {
-		return response{
-			ID:    strm.ID,
-			error: fmt.Errorf("setting write deadline failed with: %w", err),
+		return &Response{
+			ReadWriter: strm.Stream,
+			pid:        strm.ID,
+			error:      fmt.Errorf("setting write deadline failed with: %w", err),
 		}
 	}
 
 	if err := cmd.Encode(strm); err != nil {
-		return response{
-			ID:    strm.ID,
-			error: fmt.Errorf("seding command `%s(%s)` failed with: %w", cmd.Command, c.path, err),
+		return &Response{
+			ReadWriter: strm.Stream,
+			pid:        strm.ID,
+			error:      fmt.Errorf("seding command `%s(%s)` failed with: %w", cmd.Command, c.path, err),
 		}
 	}
 
 	if err := strm.SetReadDeadline(min(time.Now().Add(RecvTimeout), deadline)); err != nil {
-		return response{
-			ID:    strm.ID,
-			error: fmt.Errorf("setting read deadline failed with: %w", err),
+		return &Response{
+			ReadWriter: strm.Stream,
+			pid:        strm.ID,
+			error:      fmt.Errorf("setting read deadline failed with: %w", err),
 		}
 	}
 
 	resp, err := cr.Decode(strm)
 	if err != nil {
-		return response{
-			ID:    strm.ID,
-			error: fmt.Errorf("recv response of `%s(%s)` failed with: %w", cmd.Command, c.path, err),
+		return &Response{
+			ReadWriter: strm.Stream,
+			pid:        strm.ID,
+			error:      fmt.Errorf("recv response of `%s(%s)` failed with: %w", cmd.Command, c.path, err),
 		}
 	}
 
 	if v, k := resp["error"]; k {
-		return response{
-			ID:    strm.ID,
-			error: errors.New(fmt.Sprint(v)),
+		return &Response{
+			ReadWriter: strm.Stream,
+			pid:        strm.ID,
+			error:      errors.New(fmt.Sprint(v)),
 		}
 	}
 
-	return response{
-		ID:       strm.ID,
-		Response: resp,
+	return &Response{
+		ReadWriter: strm.Stream,
+		pid:        strm.ID,
+		Response:   resp,
 	}
 }
 
-func (c *Client) send(cmdName string, body command.Body, minStreams int) (<-chan response, error) {
-	now := time.Now()
-	ctx, ctxC := context.WithDeadline(c.ctx, now.Add(SendToPeerTimeout))
-	strmDD, _ := ctx.Deadline()
+func (c *Client) send(cmdName string, body command.Body, streams []stream, minStreams int, timeout time.Duration) (<-chan *Response, error) {
+	if timeout == 0 {
+		timeout = SendToPeerTimeout
+	}
+
+	if minStreams > MaxStreamsPerSend {
+		return nil, fmt.Errorf("threashold %d exceeds MaxStreamsPerSend", minStreams)
+	}
+
+	ctx, ctxC := context.WithTimeout(c.ctx, timeout)
+	cmdDD, _ := ctx.Deadline()
```

Comment:
move this down, not using this till later

Reply:
easy to relate to ctx here